server/src/lib/bucket.ts

```diff
@@ -1,11 +1,10 @@
 import * as path from 'path';
+import * as Random from 'random-js';
+import { S3 } from 'aws-sdk';
 
 import { map } from '../promisify';
 import { getFileExt } from './utility';
-
-const MemoryStream = require('memorystream');
-const Random = require('random-js');
-const AWS = require('./aws');
+import './aws';
```

Comment:
This is a little too bad. We have an anti-pattern where importing a file for the first time modifies the state of another global module. Making this adhere to TS standards makes the problem even more glaring.

I think the right thing to do is make our `awt.ts` module act as a proxy to the global AWS module (rather that just modifying it), and only expose s3 though that. But I'm not sure if we'll ever get to this.